AppInsights.WindowsDesktop/WindowsDesktop/WTSSessionTelemetryModule.cs

```diff
@@ -0,0 +1,100 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WTSSessionTelemetryModule"/> class.
+        /// </summary>
+        public WTSSessionTelemetryModule() : this(Application.Current.MainWindow)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WTSSessionTelemetryModule"/> class.
+        /// </summary>
+        /// <param name="window">The Window to track WTS state for</param>
+        public WTSSessionTelemetryModule(Window window)
+        {
+            if (window != null)
+            {
+                InitializeHelper(window);
+            }
+            else
+            {
+                Application.Current.Activated += Application_Activated;
+            }
+        }
+
+        private void Application_Activated(object sender, EventArgs e)
+        {
+            var window = Application.Current.MainWindow;
+            if (window != null)
+            {
+                Application.Current.Activated -= Application_Activated;
+                InitializeHelper(window);
+            }
+        }
+
+        private void InitializeHelper(Window window)
+        {
+            _helper = new WTSSessionHelper(window);
+            _helper.SessionStateChanged += WTS_SessionStateChanged;
+            _helper.RemoteControlChanged += WTS_RemoteControlChanged;
+            _helper.RemoteDesktopChanged += WTS_RemoteDesktopChanged;
+        }
+
+        /// <inheritdoc cref="ITelemetryModule.Initialize(TelemetryConfiguration)" />
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            if (!this._isInitialized)
+            {
+                lock (_lockObject)
+                {
+                    if (!_isInitialized)
+                    {
+                        _isInitialized = true;
+                        _telemetryClient = new TelemetryClient(configuration);
+                    }
+                }
+            }
+        }
+
+        private void WTS_RemoteDesktopChanged(object sender, bool isRemoteDesktop)
+        {
+            _telemetryClient?.TrackEvent("WTS Remote Desktop Session", new Dictionary<string, string>() { { "Remote Desktop", isRemoteDesktop.ToString() } });
+        }
+
+        private void WTS_RemoteControlChanged(object sender, bool isRemoteControl)
+        {
+            _telemetryClient?.TrackEvent("WTS Remote Control Session", new Dictionary<string, string>() { { "Remote Controlled", isRemoteControl.ToString() } });
+        }
+
+        private void WTS_SessionStateChanged(object sender, WTSSessionHelper.WTSSessionState state)
+        {
+            _telemetryClient?.TrackEvent("WTS Session State Changed", new Dictionary<string, string>() { { "New State", state.ToString() } });
+        }
+
+        /// <summary>
+        /// Disposing WTSSessionTelemetryModule instance.
+        /// </summary>
+        public void Dispose()
+        {
+            _helper?.Dispose();
```

Comment:
Should the dispose method here unregister the events that were hooked up earlier? It's not clear what Dispose does here

Reply:
It could but really isn't necessary, as both the helper and this instance goes out of scope together, and no events will fire after the helper got disposed..

Reply:
Okay, just making sure we're not creating a leak. It does look like there's a possible leak if the window is never activated? The module unsubscribes from the event in that case, but if it's never activated, dispose doesn't unsubscribe. Should that be tracked and handled?

Reply:
Sorry I dropped the ball on this. I'm back! How about this?